mitodl/open-discussions
# add course index page

#### Pre-Flight checklist



#### What are the relevant tickets?

#1800 

#### What's this PR do?

This PR implements the course index page, which basically shows three carousels with courses from three different APIs (new, featured, upcoming). One of the mocks it should follow is here: https://app.zeplin.io/project/5b181d3bf71408ae4d406b94/screen/5c4efd5da6396e444e8bd39a

Note that I left off the banner for the first PR, I'll do that as a follow-up (I figured this was big enough already).

#### How should this be manually tested?

Try visiting `/courses`. You should see UI like the screenshots below. You should be able to scroll the little carousels back and forth, both by dragging them and by click the 'previous' and 'next' buttons. the buttons should grey out at the two ends of the lists.

Note that to get courses to show up in the 'featured' list you'll need to go into the course admin and toggle the 'featured' flag on a few courses.

If you click on a course the course drawer should open to show some details.


#### Screenshots (if appropriate)
![courseindex_basicdrawer](https://user-images.githubusercontent.com/6207644/54293554-9177ea00-4586-11e9-9a09-a659c5619746.png)
![courseindex_basic](https://user-images.githubusercontent.com/6207644/54293556-9177ea00-4586-11e9-8c14-8ce2c0629755.png)
